src/amber/dsl/router.cr

```diff
@@ -11,13 +11,15 @@ module Amber::DSL
     macro route(verb, resource, controller, action)
       %handler = ->(context : HTTP::Server::Context, action : Symbol){
         controller = {{controller.id}}.new(context)
-        controller.run_before_filter(action)
-        content = controller.{{ action.id }}
-        controller.run_after_filter(action)
-        content
+        controller.run_before_filter(action) unless context.halt
```

Comment:
Handlers already have the ability to completely halt the request cycle but returning instead of calling next. Halt is only necessary because it all happens within the same handler proc.